### PR TITLE
Spec: detect nesting `it`/`pending` and raise compilation error

### LINKED ISSUE #7192: pending test causes incorrect test count and malformed output
Version of crystal:
```
Crystal 0.27.0 [c9d1eef8f] (2018-11-01)

LLVM: 4.0.0
Default target: x86_64-unknown-linux-gnu
```

When I mark a test as pending the count of tests and test output are malformed.

I wrote the following test.
```crystal
describe "my tests" do
  it "1 equals 1.0" do
    1.should eq 1.0
  end
end
```

Running the test correctly outputs one green test displays the expected output.
screen shot 2018-12-16 at 8 24 31 am

When I mark the test as pending, things will get interesting.
```crystal
describe "my tests" do
  it "1 equals 1.0" do
    pending "something"
    1.should eq 1.0
  end
end
```

The output shows two examples and the verbose output is malformed with the word `something` showing up twice on the first example listed and only `1.0` from the test name is shown, not the full name `1 equals 1.0`.
screen shot 2018-12-16 at 8 23 49 am

The non-verbose spec output also shows two examples, although is not malformed.
screen shot 2018-12-16 at 8 25 30 am

### PR DESCRIPTION
**Pull request by MakeNowJust**
Fixed #7192

Fixed specs following changes also.

```crystal
require "spec"

it "foo" do
  pending
end
```

```console
$ crystal run foo.cr
Error in foo.cr:3: instantiating 'it(String)'

it "foo" do
^~

in foo.cr:3: cannot nest 'pending' of spec

it "foo" do
^~

```

NOTE: Reported location looks wrong due to #7147.

### CODE REVIEW

File: spec/std/colorize_spec.cr [hunk 1]
```diff
@@ -17,116 +17,116 @@ private class ColorizeToS
   end
 end
 
-describe "colorize" do
+describe "colorize_wrap" do
```
Comment (Sija) [inline spec/std/colorize_spec.cr]: it's still about `colorize`, `colorize_wrap` is just an internal helper.

File: spec/std/colorize_spec.cr [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require "spec"
 require "colorize"
 
-private def colorize(obj, *args)
+private def colorize_wrap(obj, *args)
```
Comment (Sija) [inline spec/std/colorize_spec.cr]: why?

File: spec/std/colorize_spec.cr (hunk 1 above)
Comment (MakeNowJust) [reply spec/std/colorize_spec.cr]: It is failure of `%s/colorize/colorize_wrap`. Thank you.

File: spec/std/colorize_spec.cr (hunk 2 above)
Comment (MakeNowJust) [reply spec/std/colorize_spec.cr]: `Scope.colorize` is resolved in `it` blocks before file-scope `colorize` method, so it needs to change name.

**Comment by MakeNowJust** [comment]
Indeed. And this check pass over nesting `it` in some cases.

```crystal
def nest_test
  it "nest" do
    (1 + 1).should eq(2)
  end
end

it "test" do
  nest_test
end
```

I'll change this to a runtime check. However, finding problem before running is useful in fact... Could anyone have another idea?

**Comment by MakeNowJust** [comment]
Ok, I decide to rewrite runtime-check this. Thank you @asterite, @straight-shoota, and @bcardiff.

**Comment by bcardiff** [comment]
@straight-shoota on rspec a pending with block is executed, allowing the developer to code de spec and even check that is not able to run and will fail when the code success (a bit confusing for first timers).

I think that a pending that replaces the it is enough and node/comments in the spec should be enough. I wont expect that asserting things in the middle of an in progress spec will be so much used.

**Pull request closed without merging**